### DISCUSSION TITLE
Custom theme issue

### DISCUSSION THREAD

**Discussion by AdnanKhan01** [keycloak/keycloak#29284] [Q&A]
Hi everyone,
I am using k8s to run my keycloak.  I want to apply custom theme for keycloak. I followed the official documentation on how to do that.  I opened the shell of my running pod then created a /providers directory inside /opt/jboss/keycloak. Then I copied  the jar folder that is my custom theme inside this providers folder but it does not work.
But when I just overwrite the login or any other folder within the base directory, the changes are made but the whole CSS is disturbed. I get a big custom logo that fills the entire screen.
Help is much appreciated

**Comment by darius-m** [comment]
First check the console and network tabs of the developer tools in your browser. If you can set the theme to be used in the admin console, it means that Keycloak has loaded the theme, but something goes wrong when loading it. The browser likely fails to load the CSS files because of some missing or misconfigured dependencies in the `theme.properties` file.

I remember having an issue with setting the `styles` and `stylesCommon` files in the past, where the CSS from the parent were not loading. If I recall correctly, I believe the intention was to have `stylesCommon` keep track of the files in the default theme and you could just add your files to `styles`, but something went wrong. The workaround in our case was adding *all* CSS files in the `styles` variable (basically copying the `styles` variable from the parent and adding our override files at the end of the list), but hopefully that's no longer needed (to be fair it's not that much extra work since you must make sure the base theme is compatible with your changes after each upgrade regardless, so it does not bother me as much).

**Reply by codespearhead** [reply]
> created a /providers directory inside /opt/jboss/keycloak. Then I copied the jar folder that is my custom theme inside this providers folder

[Try putting it in /opt/jboss/keycloak/themes](https://www.keycloak.org/docs/latest/server_development/#_themes#:~:text=themes/mytheme) instead. Additionally, you might want to apply custom themes via [initContainers](https://stackoverflow.com/a/71567899) and [command flags](https://github.com/keycloak/keycloak/issues/14071#issuecomment-1292253057) instead of manually copying it into the container.
